pkg/apiserver/apiserver.go

```diff
@@ -162,6 +181,42 @@ func (c completedConfig) New() (*AdmissionServer, error) {
 		}
 	}
 
+	for _, versionMap := range conversionHooksByGroupThenVersion(c.ExtraConfig.ConversionHooks...) {
+		// TODO we're going to need a later k8s.io/apiserver so that we can get discovery to list a different group version for
+		// our endpoint which we'll use to back some custom storage which will consume the AdmissionReview type and give back the correct response
+		apiGroupInfo := genericapiserver.APIGroupInfo{
+			VersionedResourcesStorageMap: map[string]map[string]rest.Storage{},
+			// TODO unhardcode this.  It was hardcoded before, but we need to re-evaluate
+			OptionsExternalVersion: &schema.GroupVersion{Version: "v1"},
+			Scheme:                 Scheme,
+			ParameterCodec:         metav1.ParameterCodec,
+			NegotiatedSerializer:   Codecs,
+		}
+
+		for _, conversionHooks := range versionMap {
+			for i := range conversionHooks {
+				conversionHook := conversionHooks[i]
+				conversionResource, _ := conversionHook.Resource()
+				conversionVersion := conversionResource.GroupVersion()
+
+				// just overwrite the groupversion with a random one.  We don't really care or know.
+				apiGroupInfo.PrioritizedVersions = appendUniqueGroupVersion(apiGroupInfo.PrioritizedVersions, conversionVersion)
+
+				conversionReview := conversionreview.NewREST(conversionHook.Conversion)
+				v1alpha1storage, ok := apiGroupInfo.VersionedResourcesStorageMap[conversionVersion.Version]
+				if !ok {
+					v1alpha1storage = map[string]rest.Storage{}
+				}
+				v1alpha1storage[conversionResource.Resource] = conversionReview
+				apiGroupInfo.VersionedResourcesStorageMap[conversionVersion.Version] = v1alpha1storage
+			}
+		}
+
+		if err := s.GenericAPIServer.InstallAPIGroup(&apiGroupInfo); err != nil {
```

Comment:
Come to think, this may need some re-working as if a validating webhook and a converision webhook share an API group, I don't think the GenericAPIServer will handle two registrations for the same group gracefully...

Reply:
Fixed, I think 😄